elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideDynamicConfiguration.java

```diff
@@ -0,0 +1,164 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateProperties;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateProperties.Naming;
+import org.springframework.boot.autoconfigure.orm.jpa.JpaProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.persistenceunit.PersistenceUnitManager;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.persistence.Entity;
+import javax.persistence.spi.PersistenceUnitInfo;
+import javax.sql.DataSource;
+
+/**
+ * Dynamic Configuration For Elide Services. Override any of the beans (by
+ * defining your own) and setting flags to disable in properties to change the
+ * default behavior.
+ */
+
+@Slf4j
+@Configuration
+@EnableConfigurationProperties(ElideConfigProperties.class)
+@ConditionalOnExpression("${elide.dynamic-config.enabled:false}")
+public class ElideDynamicConfiguration {
+
+    public static final String HIBERNATE_DDL_AUTO = "hibernate.hbm2ddl.auto";
+    public static final String HIBERNATE_PHYSICAL_NAMING = "hibernate.physical_naming_strategy";
+    public static final String HIBERNATE_IMPLICIT_NAMING = "hibernate.implicit_naming_strategy";
+    public static final String HIBERNATE_ID_GEN_MAPPING = "hibernate.use-new-id-generator-mappings";
+
+    /**
+     * Configure factory bean to create EntityManagerFactory for Dynamic Configuration.
+     * @param source :DataSource for JPA
+     * @param jpaProperties : JPA Config Properties
+     * @param hibernateProperties : Hibernate Config Properties
+     * @param dynamicCompiler : ElideDynamicEntityCompiler
+     * @return LocalContainerEntityManagerFactoryBean bean
+     */
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory (
+            DataSource source,
+            JpaProperties jpaProperties,
+            HibernateProperties hibernateProperties,
+            ObjectProvider<ElideDynamicEntityCompiler> dynamicCompiler) {
+
+            //Map for Persistent Unit properties
+            Map<String, Object> puiPropertyMap = new HashMap<>();
+
+            //Bind entity classes from classpath to Persistence Unit
+            ArrayList<Class> bindClasses = new ArrayList<>();
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(Entity.class));
+
+            //Bind FromTable/FromSubSelect classes from classpath to Persistence Unit
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromTable.class));
```

Comment:
FromTable and FromSubquery are not JPA Entities - why are we binding them to the Persistent Unit?